configuration/pih/liquibase/sql/create_functions.sql

```diff
@@ -835,6 +857,83 @@ BEGIN
 
     RETURN enc_id_out;
 
+END
+#
+/*
+This function accepts a patient id, source & code of a concept and a locale
+It will return the single, latest value coded of the obs of that concept in the locale
+*/
+DROP FUNCTION IF EXISTS latest_obs_value_coded;
+#
+CREATE FUNCTION latest_obs_value_coded(_patient_id int(11), _source varchar(50), _term varchar(255), _locale varchar(50))
```

Comment:
This is fine I guess, but would be more re-usable to just have 2 functions like:

concept_name(_concept_id, _locale); -> returns String name
latest_obs_value_coded(_patient_id, _source, _term); -> returns int concept_id

and then compose these together as needed like:

concept_name(latest_obs_value_coded(_patient_id, _source, _term), _locale);